src/liquid/components/ld-tabs/test/ld-tabs.spec.ts

```diff
@@ -0,0 +1,665 @@
+      expect(ldTabpanel0.classList.contains('ld-tabpanel--hidden')).toEqual(
+        false
+      )
+
+      expect(tabBtn2.getAttribute('aria-selected')).toEqual(null)
+      expect(tabBtn2.getAttribute('tabindex')).toEqual('-1')
+
+      const ldTabpanel2 = ldTabpanels[2]
+      expect(ldTabpanel2.classList.contains('ld-tabpanel--hidden')).toEqual(
+        true
+      )
+
+      ldTabItems[2].scrollIntoView = jest.fn()
+      const spyScrollIntoView = jest.spyOn(ldTabItems[2], 'scrollIntoView')
+
+      tabBtn2.dispatchEvent(new Event('click'))
+      await page.waitForChanges()
+
+      expect(tabBtn0.getAttribute('aria-selected')).toEqual(null)
+      expect(tabBtn0.getAttribute('tabindex')).toEqual('-1')
+
+      expect(ldTabpanel0.classList.contains('ld-tabpanel--hidden')).toEqual(
+        true
+      )
+
+      expect(tabBtn2.getAttribute('aria-selected')).toEqual('true')
+      expect(tabBtn2.getAttribute('tabindex')).toEqual(null)
+
+      expect(ldTabpanel2.classList.contains('ld-tabpanel--hidden')).toEqual(
+        false
+      )
+
+      expect(spyScrollIntoView).toHaveBeenCalled()
+    })
+
+    it('changes tab via click with no preselected tab', async () => {
+      const page = await newSpecPage({
+        components,
+        html: `
+        <ld-tabs>
+          <ld-tablist>
+            <ld-tab>Fruits</ld-tab>
+            <ld-tab>Vegetables</ld-tab>
+            <ld-tab>Nuts</ld-tab>
+            <ld-tab disabled>Grain</ld-tab>
+          </ld-tablist>
+          <ld-tabpanellist>
+            <ld-tabpanel>Apple, orange, banana</ld-tabpanel>
+            <ld-tabpanel>Potato, cucumber, tomato</ld-tabpanel>
+            <ld-tabpanel>Walnut, chestnut, strawberry</ld-tabpanel>
+            <ld-tabpanel>Maize, rice, soybeans, wheat</ld-tabpanel>
+          </ld-tabpanellist>
+        </ld-tabs>
+      `,
+      })
+      const ldTabs = page.root
+      const ldTablist = ldTabs.querySelector('ld-tablist')
+
+      const ldTabItems = ldTablist.querySelectorAll('ld-tab')
+      expect(ldTabItems.length).toEqual(4)
+
+      const ldTabpanellist = ldTabs.querySelector('ld-tabpanellist')
+      const ldTabpanels = ldTabpanellist.querySelectorAll('ld-tabpanel')
+      expect(ldTabpanels.length).toEqual(4)
+
+      const tabBtn0 = ldTabItems[0].querySelector('button')
+      const tabBtn2 = ldTabItems[2].querySelector('button')
+
+      expect(tabBtn0.getAttribute('aria-selected')).toEqual(null)
+      expect(tabBtn0.getAttribute('tabindex')).toEqual('-1')
+
+      const ldTabpanel0 = ldTabpanels[0]
+      expect(ldTabpanel0.classList.contains('ld-tabpanel--hidden')).toEqual(
+        true
+      )
+
+      expect(tabBtn2.getAttribute('aria-selected')).toEqual(null)
+      expect(tabBtn2.getAttribute('tabindex')).toEqual('-1')
+
+      const ldTabpanel2 = ldTabpanels[2]
+      expect(ldTabpanel2.classList.contains('ld-tabpanel--hidden')).toEqual(
+        true
+      )
+
+      ldTabItems[2].scrollIntoView = jest.fn()
```

Comment:
Shouldn't the assertions inside this test before the current line be already tested in the previous unit test? Maybe we can reduce the test to only assert everything that follows this line. If a previous assertion fails, the previous test will fail, as well.

Reply:
I have removed the redundant assertions.